File: chrome/browser/resources/hotword/metrics.js
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

cr.define('hotword.metrics', function() {
  'use strict';

  /**
   * Helper function to record enum values in UMA.
   * @param {!string} name
   * @param {!number} value
   * @param {!number} maxValue
   */
  function recordEnum(name, value, maxValue) {
    var metricDesc = {
      'metricName': name,
      'type': chrome.metricsPrivate.MetricTypeType.HISTOGRAM_LINEAR,
      'min': 1,
      'max': maxValue,
      'buckets': maxValue + 1
    };
    chrome.metricsPrivate.recordValue(metricDesc, value);
  }

  return {
    recordEnum: recordEnum
  };
});

File: device/fido/enclave/metrics.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "device/fido/enclave/metrics.h"

#include "base/metrics/histogram_functions.h"

namespace device::enclave {

void RecordEvent(Event event) {
  base::UmaHistogramEnumeration("WebAuthentication.EnclaveEvent", event);
}

}  // namespace device::enclave
